lms/djangoapps/verify_student/api.py

```diff
@@ -33,3 +40,76 @@ def send_approval_email(attempt):
     else:
         email_context = {'user': attempt.user, 'expiration_datetime': expiration_datetime.strftime("%m/%d/%Y")}
         send_verification_approved_email(context=email_context)
+
+
+def create_verification_attempt(user, name, status, expiration_datetime=None):
+    """
+    Create a verification attempt.
+
+    This method is intended to be used by IDV implementation plugins to create VerificationAttempt instances.
+
+    Args:
+        user (User): the user (usually a learner) performing the verification attempt
+        name (string): the name being ID verified
+        status (string): the initial status of the verification attempt
+        expiration_datetime (datetime, optional): When the verification attempt expires. Defaults to None.
+
+    Returns:
+        id (int): The id of the created VerificationAttempt instance
+    """
+    verification_attempt = VerificationAttempt.objects.create(
+        user=user,
+        name=name,
+        status=status,
+        expiration_datetime=expiration_datetime,
+    )
+
+    return verification_attempt.id
+
+
+def update_verification_attempt(attempt_id, name=None, status=None, expiration_datetime=None):
+    """
+    Update a verification attempt.
+
+    This method is intended to be used by IDV implementation plugins to update VerificationAttempt instances.
+
+    Arguments:
+        * attempt_id (int): the verification attempt id of the attempt to update
+        * name (string, optional): the new name being ID verified
+        * status (string, optional): the new status of the verification attempt
+        * expiration_datetime (datetime, optional): The new expiration date and time
+
+    Returns:
+        * None
+    """
+    try:
+        attempt = VerificationAttempt.objects.get(id=attempt_id)
+    except VerificationAttempt.DoesNotExist:
+        log.error(
+            f'VerificationAttempt with id {attempt_id} was not found '
+            f'when updating the attempt to status={status}',
+        )
+        raise
+
+    if name is not None:
+        attempt.name = name
+
+    if status is not None:
+        attempt.status = status
+
+        status_list = [attr for attr in dir(VerificationAttemptStatus) if not attr.startswith('__')]
+        if status not in status_list:
+            log.error(
+                'Attempted to call update_verification_attempt called with invalid status: %(status)s. '
+                'Status must be one of: %(status_list)s',
+                {
+                    'status': status,
+                    'status_list': VerificationAttempt.STATUS_CHOICES,
+                },
+            )
+            raise VerificationAttemptInvalidStatus
+
+    if expiration_datetime is not None:
```

Comment:
Is it the case that if `VerificationAttempt.expiration_datetime` is `None`, that means "this verification never expires"? If that's the case, then if a `expiration_datetime` was set to be sometime next year, and I wanted to update it to never expire, is that possible?

Reply:
If the value of the `expiration_datetime` parameter sent to this function is `None`, then the `expiration_datetime` of the IDV attempt will _not be updated_. This is to maintain that all IDV attempts will expire/retire at some point (which at least for IDV attempts via Persona, should be 14 days).

Reply:
That makes sense to me, thank you for clarifying that! I'll push this change soon.